### PR TITLE
Fusion update - Removes volume dependence and patches the toroidal scale exploit

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Removes dependence on scaling the mole counts of the fusion reaction by volume. Instead, it takes in the mole count of the input gases and scales off of that, preventing an exploit where you could produce an insane amount of matter from nothing for virtually zero cost. It also makes starting fusion a bit easier while making the ascent up to the maximum temperature cap a bit harder by removing the situations where a high instability reaction tick kicks down both plasma and hydrogen counts to tiny, tiny amounts slightly above the mole threshold. To partially compensate for this, energy production is doubled per unit of plasma burned. There is also now a molar softcap of 5000 moles. Going below it mitigates the thermal effects of high instability reactions, but going above it basically functions the same as before. 

Water vapor is now used in the fusion reaction and has been removed from the pool of waste gases. Unfortunately this means only carbon dioxide is emitted, but I see no alternative aside from making gases specifically for fusion that have no interactions with other gases in order to preserve mole counts.

Warning: Math ahead.

Specifically, it removes the toroidal_scale variable which is based off of volume and just adds up mole counts. Before this, the maximum amount of gas of the rotors was equivalent to 4pi radians * toroidal scale, which was fine since toroidal scale was independent of gas amounts. But now, a third rotor has had to be added to make up the difference for when rotor angular positions did not add up to 4pi radians, which is basically all the time. With this, the mole count information from the first tick is preserved throughout ticks regardless of the angular positions of the 2 primary rotors are. 

This also preserves the cap on the 2 rotors individually being capped to 2pi radians by scaling total gas amounts to 4pi radians. However, in instances where any of the 2 rotors exceed 2pi radians, such as during the initial gasmix before fusion can tick, any amount in excess of 2pi radians is instantly converted to energy with the rotor being kicked down to the 2pi radian max before the proper k-parameter kick can begin. 

The doubling of energy is a necessity since it basically halves the range at which mole counts can vary. The molar softcap was introduced as the average energy of a fusion tick across all rotor positions was dependent entirely on volume and would guarantee endothermic reactions above values of 1367L, or somewhere there, and all reactions below that were eventually guaranteed to yield positive energy gain. Fixing this probabilistic guarantee is significantly more effort than it is worth so I bumped it up an order of magnitude by lowering the instability exponent and using a 5000 mole softcap. This should make it much, much easier to do fusion than it currently is.

Lastly, a note to anyone porting this. Do not port this if you want to keep the high temperature cap of e31, this was specifically designed around TG's e8 temperature cap. The specific effect that allowed thermals to go up orders of magnitude due to sudden deletion of gases (when both rotors get kicked down to the 0~0.25pi position, usually), has been almost entirely mitigated and is a necessary part of the transition away from volume. There is no way around this.

## Why It's Good For The Game

Patches an exploit and makes starting fusion easier, but makes reaching higher heat scales a harder task.

## Changelog
:cl:
tweak: Fusion is now different. A bit easier to start, but a bit harder to make it reach the maximum.
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -455,7 +455,7 @@ nobiliumsuppression = INFINITY
 	var/reaction_energy = 0 //Reaction energy can be negative or positive, for both exothermic and endothermic reactions.
 	var/initial_plasma = cached_gases[/datum/gas/plasma][MOLES]
 	var/initial_hydrogen = cached_gases[/datum/gas/hydrogen][MOLES]
-	//var/scale_factor = (air.volume)/(PI) //We scale it down by volume/Pi because for fusion conditions, moles roughly = 2*volume, but we want it to be based off something constant between reactions.
+	air.assert_gases(/datum/gas/water_vapor)v
```
Comment (Ghilker) [inline code/modules/atmospherics/gasmixtures/reactions.dm:458]: there's a v here

Comment (AutisticFroggy) [reply code/modules/atmospherics/gasmixtures/reactions.dm:458]: bruh

**Comment by Moja10** [comment]
Noooooo you can't just make a PR that not even the maintainers can understand noooo!!!
haha autism go brrrrrr

**Comment by optimumtact** [comment]
Well well, look who it is

**Comment by Moja10** [comment]
told you the maintainers don't understand it 😂

**Comment by AutisticFroggy** [comment]
Still testing this, some extreme edge cases where all three rotors aligned caused weird thermal effects. Need to iron out the math a bit more.

**Comment by AutisticFroggy** [comment]
oh god something broke

**Comment by AutisticFroggy** [comment]
Closing this until I finish some IRL stuff instead of waiting for a maint to come by and close it. I'll finish this some time.

**Pull request closed without merging**